### PR TITLE
BEE CI + Integration Tests

### PR DESCRIPTION
**Pull request by jtronge**
This PR addresses issue #419. I've added two workflows here, one for pylama, and the other for an integration test of BEE. We'll probably want another for the unit tests/general testing matrix.

The integration test is designed to set up and install BEE, launch the CLAMR workflow and then report back on whether it fails or not. Later on we may want to add more workflows here.

### CODE REVIEW

File: .github/workflows/ci-workflow.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: Integration workflow
+
+on:
+  push:
+    branches:
+      - ci-workflow
```
Comment (jtronge) [inline .github/workflows/ci-workflow.yml]: This is not set correctly yet. I'm not sure when exactly we will want to run the integration test. On opening PRs to develop? On each change?

**Comment by jtronge** [comment]
For the integration test which runs the clamr workflow, I managed to better separate the output into multiple steps so that it's a little bit easier to look over. There is still a lot there but it's much less than before, and we'll probably need it for when a workflow fails.

I've also added a stub unit test workflow. Right now it just calls pytest on .py files in the `src/beeflow/tests` directory, but it may need some more work.

File: .github/workflows/ci-workflow.yml (hunk 1 above)
Comment (pagrubel) [reply .github/workflows/ci-workflow.yml]: Perhaps on opening PR's and upon removing WIP if possible.

File: src/beeflow/data/cwl/bee_workflows/clamr-ci/clamr_job.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Inputs for CLAMR
+# /clamr/CLAMR-master/clamr_cpuonly -n 32 -l 3 -t 5000 -i 10 -g 25 -G png
+
+grid_resolution: 32
+max_levels: 3
+time_steps: 5000
```
Comment (pagrubel) [inline src/beeflow/data/cwl/bee_workflows/clamr-ci/clamr_job.yml]: Should we shorten this for the integration test, right now I think it takes a couple of minutes on fog to run the clamr step, not sure how long on CI. CI should be as short as possible.

**Review by pagrubel — DISMISSED** [review]

**Review by pagrubel — APPROVED** [review]

**Comment by pagrubel** [comment]
This sounds like a good start. I’m thinking of some more rigorous tests but thing they might be more suited to periodic testing, nightly or even less frequently or on demand.

Pat

From: jtronge ***@***.***>
Reply-To: lanl/BEE_Private ***@***.***>
Date: Wednesday, June 29, 2022 at 9:13 AM
To: lanl/BEE_Private ***@***.***>
Cc: "Grubel, Patricia A" ***@***.***>, Review requested ***@***.***>
Subject: [EXTERNAL] [lanl/BEE_Private] BEE CI + Integration Tests (PR #478)


This PR addresses issue #419<https://urldefense.com/v3/__https:/github.com/lanl/BEE_Private/issues/419__;!!Bt8fGhp8LhKGRg!AQT45Mxjb5_SeYMhV3KmX8YykChHzUTaVdH8FMlXC1Clehkpt8wPvANVPr1V3h8La57OfwJy_VE2wJO5V-7Vc_tMmA$>. I've added two workflows here, one for pylama, and the other for an integration test of BEE. We'll probably want another for the unit tests/general testing matrix.

The integration test is designed to set up and install BEE, launch the CLAMR workflow and then report back on whether it fails or not. Later on we may want to add more workflows here.

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/lanl/BEE_Private/pull/478<https://urldefense.com/v3/__https:/github.com/lanl/BEE_Private/pull/478__;!!Bt8fGhp8LhKGRg!AQT45Mxjb5_SeYMhV3KmX8YykChHzUTaVdH8FMlXC1Clehkpt8wPvANVPr1V3h8La57OfwJy_VE2wJO5V-6nJX1X7w$>

Commit Summary

  *   658ef6b<https://urldefense.com/v3/__https:/github.com/lanl/BEE_Private/pull/478/commits/658ef6b68d7caf41c48cc2951a8160eccd72837b__;!!Bt8fGhp8LhKGRg!AQT45Mxjb5_SeYMhV3KmX8YykChHzUTaVdH8FMlXC1Clehkpt8wPvANVPr1V3h8La57OfwJy_VE2wJO5V-73Do8Hzg$> Add CI script, workflow and Pylama lint
  *   72ad170<https://urldefense.com/v3/__https:/github.com/lanl/BEE_Private/pull/478/commits/72ad170045571715af2900fc67c7973bcfdee8bc__;!!Bt8fGhp8LhKGRg!AQT45Mxjb5_SeYMhV3KmX8YykChHzUTaVdH8FMlXC1Clehkpt8wPvANVPr1V3h8La57OfwJy_VE2wJO5V-7rmvOSmg$> Add return code

File Changes

(11 files<https://urldefense.com/v3/__https:/github.com/lanl/BEE_Private/pull/478/files__;!!Bt8fGhp8LhKGRg!AQT45Mxjb5_SeYMhV3KmX8YykChHzUTaVdH8FMlXC1Clehkpt8wPvANVPr1V3h8La57OfwJy_VE2wJO5V-5dXkSd1g$>)

  *   A .github/workflows/ci-workflow.yml<https://urldefense.com/v3/__https:/github.com/lanl/BEE_Private/pull/478/files*diff-e42b2b9794d0333e41c8e218ece5897ea7395746b5e07d08fb85b53e36d15043__;Iw!!Bt8fGhp8LhKGRg!AQT45Mxjb5_SeYMhV3KmX8YykChHzUTaVdH8FMlXC1Clehkpt8wPvANVPr1V3h8La57OfwJy_VE2wJO5V-4rIs0pHA$> (19)
  *   A .github/workflows/pylama.yml<https://urldefense.com/v3/__https:/github.com/lanl/BEE_Private/pull/478/files*diff-2cd885d885c9829f2f1a65af3bea869c4b79d53969d63a5dbae598cd7cc02159__;Iw!!Bt8fGhp8LhKGRg!AQT45Mxjb5_SeYMhV3KmX8YykChHzUTaVdH8FMlXC1Clehkpt8wPvANVPr1V3h8La57OfwJy_VE2wJO5V-4qMOM6RQ$> (20)
  *   A ci/integration_test.py<https://urldefense.com/v3/__https:/github.com/lanl/BEE_Private/pull/478/files*diff-05df4931de530873a58f80f3741174fa1cf23185d3730360f8f90e15e1becd0c__;Iw!!Bt8fGhp8LhKGRg!AQT45Mxjb5_SeYMhV3KmX8YykChHzUTaVdH8FMlXC1Clehkpt8wPvANVPr1V3h8La57OfwJy_VE2wJO5V-6VVVxvkw$> (133)
  *   A ci/integration_test.sh<https://urldefense.com/v3/__https:/github.com/lanl/BEE_Private/pull/478/files*diff-87dc7ddfab13d0db4babfbcd1297d76c514aaf4d8f5c67be8230197aecd801e6__;Iw!!Bt8fGhp8LhKGRg!AQT45Mxjb5_SeYMhV3KmX8YykChHzUTaVdH8FMlXC1Clehkpt8wPvANVPr1V3h8La57OfwJy_VE2wJO5V-6l_vJ35Q$> (224)
  *   M src/beeflow/client/bee_client.py<https://urldefense.com/v3/__https:/github.com/lanl/BEE_Private/pull/478/files*diff-d7321afbdf20471ac9580e53149b92feda68402e018a727210bd7c666f123c4f__;Iw!!Bt8fGhp8LhKGRg!AQT45Mxjb5_SeYMhV3KmX8YykChHzUTaVdH8FMlXC1Clehkpt8wPvANVPr1V3h8La57OfwJy_VE2wJO5V-4X2-qtcw$> (26)
  *   A src/beeflow/data/cwl/bee_workflows/clamr-ci/README.md<https://urldefense.com/v3/__https:/github.com/lanl/BEE_Private/pull/478/files*diff-e59ea4fa94dc5711282a7aebbc4acd63a095e2aac7d6176b992a2e784763f14c__;Iw!!Bt8fGhp8LhKGRg!AQT45Mxjb5_SeYMhV3KmX8YykChHzUTaVdH8FMlXC1Clehkpt8wPvANVPr1V3h8La57OfwJy_VE2wJO5V-7Ibb6h1w$> (10)
  *   A src/beeflow/data/cwl/bee_workflows/clamr-ci/clamr.cwl<https://urldefense.com/v3/__https:/github.com/lanl/BEE_Private/pull/478/files*diff-a7fa505554f7971001983a42cc6cc8362955a395b560b5bfa8289730b0f50dac__;Iw!!Bt8fGhp8LhKGRg!AQT45Mxjb5_SeYMhV3KmX8YykChHzUTaVdH8FMlXC1Clehkpt8wPvANVPr1V3h8La57OfwJy_VE2wJO5V-5plqjZOg$> (83)
  *   A src/beeflow/data/cwl/bee_workflows/clamr-ci/clamr_job.json<https://urldefense.com/v3/__https:/github.com/lanl/BEE_Private/pull/478/files*diff-aefe84b56c6bcece45ca3f8199c7441d216767c0365ccf7ef43465c65e4ba7b1__;Iw!!Bt8fGhp8LhKGRg!AQT45Mxjb5_SeYMhV3KmX8YykChHzUTaVdH8FMlXC1Clehkpt8wPvANVPr1V3h8La57OfwJy_VE2wJO5V-56acbvSg$> (13)
  *   A src/beeflow/data/cwl/bee_workflows/clamr-ci/clamr_job.yml<https://urldefense.com/v3/__https:/github.com/lanl/BEE_Private/pull/478/files*diff-c99516daaca9a49e553950d6388a7fd5850b79b908585ccc95aa9cfd86f3c4b4__;Iw!!Bt8fGhp8LhKGRg!AQT45Mxjb5_SeYMhV3KmX8YykChHzUTaVdH8FMlXC1Clehkpt8wPvANVPr1V3h8La57OfwJy_VE2wJO5V-4tlJJBXg$> (18)
  *   A src/beeflow/data/cwl/bee_workflows/clamr-ci/clamr_wf.cwl<https://urldefense.com/v3/__https:/github.com/lanl/BEE_Private/pull/478/files*diff-6d8048830c346fb5ece9b8a0b60f816cae2f1d1f2a76396a089ac754e198386b__;Iw!!Bt8fGhp8LhKGRg!AQT45Mxjb5_SeYMhV3KmX8YykChHzUTaVdH8FMlXC1Clehkpt8wPvANVPr1V3h8La57OfwJy_VE2wJO5V-4S3TCb5Q$> (81)
  *   A src/beeflow/data/cwl/bee_workflows/clamr-ci/ffmpeg.cwl<https://urldefense.com/v3/__https:/github.com/lanl/BEE_Private/pull/478/files*diff-e28ea0bec07decd8134f8e0be571a223eadc11b888ba5b56c500b1a6c51fe40b__;Iw!!Bt8fGhp8LhKGRg!AQT45Mxjb5_SeYMhV3KmX8YykChHzUTaVdH8FMlXC1Clehkpt8wPvANVPr1V3h8La57OfwJy_VE2wJO5V-6f1ELAIg$> (44)

Patch Links:

  *   https://github.com/lanl/BEE_Private/pull/478.patch<https://urldefense.com/v3/__https:/github.com/lanl/BEE_Private/pull/478.patch__;!!Bt8fGhp8LhKGRg!AQT45Mxjb5_SeYMhV3KmX8YykChHzUTaVdH8FMlXC1Clehkpt8wPvANVPr1V3h8La57OfwJy_VE2wJO5V-5M1h79Nw$>
  *   https://github.com/lanl/BEE_Private/pull/478.diff<https://urldefense.com/v3/__https:/github.com/lanl/BEE_Private/pull/478.diff__;!!Bt8fGhp8LhKGRg!AQT45Mxjb5_SeYMhV3KmX8YykChHzUTaVdH8FMlXC1Clehkpt8wPvANVPr1V3h8La57OfwJy_VE2wJO5V-70eWxoNg$>

**Pull request merged**